### ISSUE TITLE
Fix for updating to Next 14.0

### ISSUE THREAD

**Issue by martinez-hugo** [resendlabs/resend-node#244]
Hello !

I've just update my project to Next v14 but I had this error generated by upgrading : 

```
Failed to compile.
--
08:53:51.047 |  
08:53:51.047 | ./node_modules/@react-email/render/dist/index.mjs
08:53:51.048 | Attempted import error: 'renderToReadableStream' is not exported from 'react-dom/server' (imported as 'renderToReadableStream').
08:53:51.048 |  
08:53:51.048 | Import trace for requested module:
08:53:51.048 | ./node_modules/@react-email/render/dist/index.mjs
08:53:51.048 | ./node_modules/resend/build/src/resend.js
08:53:51.048 | ./node_modules/resend/build/src/index.js
08:53:51.048 | ./src/app/api/send/route.tsx
```

Do you think that you can fix this error on your package ?

Thanks for all

**Comment by zenorocha** [comment]
just so you all know, we're currently working on adding support for Next 14

**Comment by russelh15** [comment]
Any updates on this?

**Comment by T04435** [comment]
I have updated to "resend": "^2.0.0-canary.0", and it is fixed.

When can we have a stable v2?

**Comment by bukinoshita** [comment]
We didn't find a fix for react-email render yet, but still working on this. The `resend@2.0.0-canary.0` uses the `@react-email/render@v0.0.9-canary.0` version which fixes the issue for Next.js 14, but breaks for Next.js 13. Once we fix that issue for both versions, we're going to update the Resend package.

**Comment by bukinoshita** [comment]
For reference, out tests with React Email passed with the latest canary version (https://github.com/resendlabs/react-email/issues/989#issuecomment-1787746290). We'll be working on comparability with Resend now

**Comment by KingXP-Pythoner** [comment]
I'm not using the render/renderAsync functions and still running into the issue. I've also tried `resend@2.0.0-canary.0`

**Comment by bukinoshita** [comment]
> I'm not using the render/renderAsync functions and still running into the issue. I've also tried `resend@2.0.0-canary.0`

Can you share your setup and how you're doing? I can include in the tests once I start working on the resend-node

**Comment by bukinoshita** [comment]
> I'm not using the render/renderAsync functions and still running into the issue. I've also tried `resend@2.0.0-canary.0`

Can you try the `2.0.0-canary.2`?

**Comment by T04435** [comment]
@bukinoshita Tested with `2.0.0-canary.2` it also works.

in terms of setup:
- pnpm workspaces
  - apps
    - web(Next14) <- use resend here
  - packages
    - resend 
      - lib/index.ts => resend class with functions to send email Templates
      - emails => react-email templates

all our packages are close to latest version(weekly updated)

**Comment by KingXP-Pythoner** [comment]
> > I'm not using the render/renderAsync functions and still running into the issue. I've also tried `resend@2.0.0-canary.0`
> 
> Can you share your setup and how you're doing? I can include in the tests once I start working on the resend-node

Hi, thanks for the response, I just tried with `2.0.0-canary.2` and it didn't work then i realised the `@react-email/components` was outdated so i updated that and it works now! Thanks a lot

**Comment by Jordaneisenburger** [comment]
@bukinoshita the resend@2.0.0-canary.2 version fixed the issue for me, thanks a lot!

**Comment by T04435** [comment]
@bukinoshita 👍🏼

**Comment by petergombos** [comment]
Works fine in api routes but it fails in Server actions:

<img width="534" alt="Screenshot 2023-11-19 at 21 54 59" src="https://github.com/resendlabs/resend-node/assets/1838896/1d70b848-1127-429d-a291-71117df88162">
<img width="1016" alt="Screenshot 2023-11-19 at 21 45 44" src="https://github.com/resendlabs/resend-node/assets/1838896/c683c48a-439a-43da-a792-085e5cc18c35">

**Comment by YuBenhao** [comment]
> We didn't find a fix for react-email render yet, but still working on this. The `resend@2.0.0-canary.0` uses the `@react-email/render@v0.0.9-canary.0` version which fixes the issue for Next.js 14, but breaks for Next.js 13. Once we fix that issue for both versions, we're going to update the Resend package.

Works for me!

**Comment by kingmuffy** [comment]
> this issue is still not fixed , I have the error with "@"resend": "^2.0.0-canary.2", and "@react-email/render": "^0.0.9-canary.0",
> 
> I will take off this library and use another thing , my page dosent work
> 
> ![image](https://private-user-images.githubusercontent.com/117024247/285706361-71c035b4-0313-4502-bd3e-7a538511e440.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU3NTE3NDksIm5iZiI6MTcwNTc1MTQ0OSwicGF0aCI6Ii8xMTcwMjQyNDcvMjg1NzA2MzYxLTcxYzAzNWI0LTAzMTMtNDUwMi1iZDNlLTdhNTM4NTExZTQ0MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyMFQxMTUwNDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYjQxMDY3Y2E4NWJkNjc4MmQ5MTBjZGRjMTY0MDg0NDVjYjFiODU3OTA4ZjJlNDFhYzU2Njk1MTYzZjVlZjU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hNJ2FWpi0JXWXzu9L0xpvuy98Nub1yWJG9_Yd_QWMQU)

please suggest other libraries. nothing is working for me.
